File: Bank/cpp/src/Transaction.cpp
#include "Transaction.h"

Transaction::Transaction(const std::string& date, const int amount)
{
        this->m_date = date;
        this->m_amount = amount;
}

const std::string
Transaction::date() const
{
        return this->m_date;
}

const int
Transaction::amount() const
{
        return this->m_amount;
}

const bool
Transaction::operator==(const Transaction& rhs) const
{
        return this->m_date == rhs.m_date && this->m_amount == rhs.m_amount;
}

File: Bank/java/src/main/java/com/iainhemstock/Transaction.java
package com.iainhemstock;

import java.util.Objects;

public class Transaction {
    private final TransactionAmount transactionAmount;
    private final TransactionDate date;


    public Transaction(TransactionAmount transactionAmount, TransactionDate date) {
        this.transactionAmount = transactionAmount;
        this.date = date;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Transaction that = (Transaction) o;
        return Objects.equals(transactionAmount, that.transactionAmount);
    }

    @Override
    public String toString() {
        return "Transaction{" +
                "transactionAmount=" + transactionAmount +
                '}';
    }

    public TransactionAmount amount() {
        return this.transactionAmount;
    }

    public TransactionDate date() {
        return this.date;
    }
}
